apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -114,10 +120,10 @@ public override int GetHashCode()
                 unchecked
                 {
                     return ((Credential?.GetHashCode() ?? 0) * 397) ^
-                        (Endpoint?.GetHashCode() ?? 0);
+                        (Endpoint?.GetHashCode() ?? 0) ^
+                        (AdditionalOptions?.Count.GetHashCode() ?? 0);
```

Comment:
Possibly write a better hash code here than just the count? Not sure how often we'll have keys that differ only by options...